apps/src/blockly/addons/cdoXml.js

```diff
@@ -36,10 +37,15 @@ export default function initializeBlocklyXml(blocklyWrapper) {
     return block;
   };
 
-  // Decode an XML DOM and create blocks on the workspace.
+  /**
+   * Decode an XML DOM and create blocks on the workspace while preserving the original order of blocks.
+   *
+   * @param {Blockly.Workspace} workspace - The Blockly workspace where blocks will be created.
+   * @param {Element} xml - The XML DOM containing block elements to be created on the workspace.
+   * @returns {Object[]} An array of objects containing the created blocks and their positions.
+   */
   blocklyWrapper.Xml.domToBlockSpace = function (workspace, xml) {
-    Blockly.Xml.createBlockOrderMap(xml);
```

Comment:
Yes, that was intentional! `createBlockOrderMap` returns a map, but otherwise doesn't modify the `xml` argument. I had added this line early-on when the function also had a bunch of `console.log` statements so I could check that it was working as I expected. `createBlockOrderMap` only is meant to get called in `cdoUtils.js` > `loadBlocksToWorkspace`, which happens after `domToBlockSpace` is used to convert the XML to JSON.